File: 2022-2023 ICPC Brazil Subregional Programming Contest/.F_failures/F'.py
# https://codeforces.com/gym/103960/problem/F

from typing import List, Tuple


def solve(words: List[str]) -> Tuple[str, int]:
    N: int = len(words)

    indexes: List[int] = [word.find('*') for word in words]
    candidates: List[str] = [
        words[i][:indexes[i]] + c + words[i][indexes[i]+1:] for i in range(N) for c in "abcdefghijklmnopqrstuvwxyz"]

    # Count compatibilities
    compatibilities: List[int] = [0] * len(candidates)
    for i, candidate in enumerate(candidates):
        for j in range(N):
            if candidate[:indexes[j]] == words[j][:indexes[j]] and candidate[indexes[j]+1:] == words[j][indexes[j]+1:]:
                compatibilities[i] += 1

    max_compatibilities: int = max(compatibilities)
    max_compatibilities_candidates: List[str] = [
        candidates[i] for i in range(len(candidates)) if compatibilities[i] == max_compatibilities]

    max_compatibilities_candidates.sort()
    return max_compatibilities_candidates[0], max_compatibilities

def parse() -> List[str]:
    N, _ = map(int, input().split())
    return [input() for _ in range(N)]


def main():
    words = parse()
    res = solve(words)
    print(*res)


if __name__ == '__main__':
    main()

File: 2022-2023 ICPC Brazil Subregional Programming Contest/.F_failures/F'.cpp
// https://codeforces.com/gym/103960/problem/F

#include <bits/stdc++.h>
#include <iostream>
#include <vector>

using namespace std;

typedef unsigned int uint;

// Function that returns a vector of string

static vector<string> parse(void) {
    uint N, C;
    cin >> N >> C;
    vector<string> words(N);
    for (uint i = 0; i < N; i++) {
        cin >> words[i];
    }

    return words;
}

// Comparator Function
static bool string_ord(string s1, string s2) {
    return s1 < s2;
}

static pair<string, uint> solve(vector<string> words) {
    uint N = words.size();
    uint C = words[0].size();

    vector<uint> indexes(N);
    for (uint i = 0; i < N; i++) {
        // Find the first '*' in the string
        for (uint j = 0; j < C; j++) {
            if (words[i][j] == '*') {
                indexes[i] = j;
                break;
            }
        }
    }

    vector<string> candidates(N * 26);
    for (uint i = 0; i < N; i++) {
        for (uint k = 0; k < 26; k++) {
            char c = 'a' + k;
            candidates[i * 26 + k] = words[i];
            candidates[i * 26 + k][indexes[i]] = c;
        }
    }

    vector<uint> compatibilities(N * 26, 0);
    for (uint k = 0; k < N * 26; k++) {
        string candidate = candidates[k];
        for (uint i = 0; i < N; i++) {
            string word = words[i];
            bool compatible = true;
            for (uint j = 0; j < C; j++) {
                if (word[j] != '*' && word[j] != candidate[j]) {
                    compatible = false;
                    break;
                }
            }
            if (compatible) {
                compatibilities[k]++;
            }
        }
    }

    uint max_compatibilities = 0;
    vector<uint> max_indexes = {};
    for (uint k = 0; k < N * 26; k++) {
        if (compatibilities[k] > max_compatibilities) {
            max_compatibilities = compatibilities[k];
            max_indexes = {k};
        } else if (compatibilities[k] == max_compatibilities) {
            max_indexes.push_back(k);
        }
    }

    vector<string> best_candidates = {};
    for (uint k : max_indexes) {
        best_candidates.push_back(candidates[k]);
    }

    sort(best_candidates.begin(), best_candidates.end(), string_ord);

    return {best_candidates[0], max_compatibilities};
}

int main(void) {
    vector<string> words = parse();
    pair<string, uint> result = solve(words);
    cout << result.first << ' ' << result.second;
    return EXIT_SUCCESS;
}
